unstable/lens/hash-slice.rkt

```diff
@@ -0,0 +1,26 @@
+#lang racket/base
+
+(provide hash-slice-lens)
+
+(require lens/base/main)
+(module+ test
+  (require rackunit))
+
+(define (hash-slice-lens . ks)
+  (define (get hsh)
+    (for/hash ([k (in-list ks)])
+      (values k (hash-ref hsh k))))
+  (define (set hsh hsh2)
+    (define hsh1
+      (for/fold ([hsh hsh]) ([k (in-list ks)])
+        (hash-remove hsh k)))
+    (for/fold ([hsh hsh1]) ([(k v) (in-hash hsh2)])
+      (hash-set hsh k v)))
+  (make-lens get set))
```

Comment:
It's indeed the case that `compound-hash-lens` requires you give it a hash with the same keys as the hash it views. However looking over the implementation I realize there's a bug in it relies on an implicit ordering of the keys, so you could accidentally set the wrong values with it. Will make issue and fix for that